xaitk_saliency/interfaces/vis_sal_classifier.py

```diff
@@ -0,0 +1,84 @@
+import abc
+
+import numpy as np
+from smqtk_core import Plugfigurable
+
+
+class ImageClassifierSaliencyMapGenerator(Plugfigurable):
+    """
+    Visual saliency map generation interface whose implementations transform
+    black-box image classification scores into saliency heatmaps.
+
+    This should require a sequence of per-class confidences predicted on the
+    reference image, a number of per-class confidences as predicted on
+    perturbed images, as well as the masks of the reference image perturbations
+    (as would be output from a
+    :class:`xaitk_saliency.interfaces.perturb_image.PerturbImage`
+    implementation).
+
+    Implementations should use this input to generate a visual saliency
+    heat-map for each input “class” in the input.
+    This is both an effort to vectorize the operation for optimal performance,
+    as well as to allow some algorithms to take advantage of differences in
+    classification behavior for other classes to influence heatmap generation.
+    For classifiers that generate many class label predictions, it is intended
+    that only a subset of relevant class predictions need be provided here if
+    computational performance is a consideration.
+    """
+
+    @abc.abstractmethod
+    def generate(
+        self,
+        image_conf: np.ndarray,
+        perturbed_conf: np.ndarray,
```

Comment:
Just to check my understanding- the implementation is responsible for checking that the selected class indices in image_conf and perturbed_conf align?

Reply:
I want to say that we want to convey that implementors should be able to assume they are the same dimensionality. We could employ the template method here to inject specific dimensionality assertions to make that check automatic and explicit.

Reply:
Resolved based on offline conversations.